src/client/scenes/Game.ts

```diff
@@ -161,10 +104,72 @@ export default class Game extends Phaser.Scene {
     selectionItem.setDialogBox('Press E to sit', 80)
   }
 
+  private addObjectFromTiled(
+    group: Phaser.Physics.Arcade.StaticGroup,
+    object: Phaser.Types.Tilemaps.TiledObject,
+    key: string,
+    tilesetName: string
+  ) {
+    const actualX = object.x! + object.width! * 0.5
+    const actualY = object.y! - object.height! * 0.5
+    const obj = group
+      .get(actualX, actualY, key, object.gid! - this.map.getTileset(tilesetName).firstgid)
+      .setDepth(actualY)
+    return obj
+  }
+
+  private addGroupFromTiled(
+    objectLayerName: string,
+    key: string,
+    tilesetName: string,
+    collidable: boolean
+  ) {
+    const group = this.physics.add.staticGroup()
+    const objectLayer = this.map.getObjectLayer(objectLayerName)
+    objectLayer.objects.forEach((object) => {
+      const actualX = object.x! + object.width! * 0.5
+      const actualY = object.y! - object.height! * 0.5
+      group
+        .get(actualX, actualY, key, object.gid! - this.map.getTileset(tilesetName).firstgid)
+        .setDepth(actualY)
+    })
+    if (this.myPlayer && collidable) this.physics.add.collider(this.myPlayer, group)
+  }
+
+  // function to add new player to the otherPlayer group
+  private handlePlayerJoined(newPlayer: IPlayer, key: string) {
+    if (!this.otherPlayers) {
+      this.otherPlayers = this.physics.add.group({ classType: OtherPlayer })
+    }
+    this.otherPlayers.get(newPlayer.x, newPlayer.y, 'player', key) as OtherPlayer
+  }
+
+  // function to remove the player who left from the otherPlayer group
+  private handlePlayerLeft(key: string) {
+    const player = this.getOtherPlayerByID(key)
+    if (player) {
+      this.otherPlayers?.remove(player, true, true)
+    }
+  }
+
+  // function to update target position upon receiving player updates
+  private handlePlayerUpdated(field: string, value: number | string, key: string) {
+    const player = this.getOtherPlayerByID(key) as OtherPlayer
+    if (player) {
+      player.updateTargetPosition(field, value)
+    }
+  }
+
+  private getOtherPlayerByID(key: string) {
```

Comment:
annotate the return type for this method.
something like
```ts
  private getOtherPlayerByID(key: string): OtherPlayer|undefined {
```

Reply:
Done!